### PR TITLE
RUBY-3433:  Tech Debt : remove devise and cancancan gems + remove front-office users from wcr engine

### PR DESCRIPTION
**Pull request by brujeo**
WCR - TechDebt - Remove the devise and cancancan gems from WCR engine
https://eaflood.atlassian.net/browse/RUBY-3433

- remove _devise_ gem and related code from the engine. Move still necessary functionality over to back-office
- remove _cancancan_ gem and related code from the engine.
- remove user model and related code and tests from the engine. Move still necessary functionality over to back-office
- update README

### CODE REVIEW

File: spec/models/waste_carriers_engine/conviction_sign_off_spec.rb
Comment (brujeo) [inline spec/models/waste_carriers_engine/conviction_sign_off_spec.rb:1]: this test has been moved to back-office

File: app/controllers/waste_carriers_engine/transient_registrations_controller.rb
Comment (brujeo) [inline app/controllers/waste_carriers_engine/transient_registrations_controller.rb:1]: this controller and related spec has been moved to back-office

File: spec/requests/waste_carriers_engine/transient_registrations_spec.rb
Comment (brujeo) [inline spec/requests/waste_carriers_engine/transient_registrations_spec.rb:1]: this test has been moved to back-office

File: spec/support/shared_examples/transient_registration_named_scopes.rb
Comment (brujeo) [inline spec/support/shared_examples/transient_registration_named_scopes.rb:1]: this test has been moved to back-office

File: spec/models/waste_carriers_engine/registration_spec.rb [hunk 1]
```diff
@@ -849,22 +849,19 @@ module WasteCarriersEngine
     end
 
     describe "#increment_certificate_version" do
-      let(:user) { create(:user) }
```
[the remark this answers is not in the document]
Comment (brujeo) [reply spec/models/waste_carriers_engine/registration_spec.rb:852]: All specs that still need to test logged in user have been moved over to back-office

[the remark this answers is not in the document]
Comment (brujeo) [reply spec/models/waste_carriers_engine/registration_spec.rb:852]: in this particular case, same test but with the logged in user have been added to back office

**Review by PaulDoyle-EA — APPROVED** [review]

**Pull request merged**